### ISSUE TITLE
Global port forwards in okteto manifest

### ISSUE THREAD

**Issue by pchico83** [okteto/okteto#2709]
**Is your feature request related to a problem? Please describe.**
I have an okteto manifest with several dev containers. In every dev container, I need to expose my database ports to access my database on localhost. This makes it impossible to run two dev containers in parallel because of port collisions.

This problem is especially relevant when using hybrid development with `reverse` because I need to be able to access all my third-party dependencies on localhost.

**Describe the solution you'd like**
A section in the okteto manifest for global forwards, something like this:

```
forward:
  - 6379:redis:6379
  - 27017:mongo:27017
  - 3306:mysql:3306
  - 9092:kafka:9092
dev:
  api:
    reverse:
      - 3000:3000

  frontend:
    reverse:
      - 8080:3000
```

instead of what we have today:
```
dev:
  api:
    forward:
      - 6379:redis:6379
      - 27017:mongo:27017
      - 3306:mysql:3306
      - 9092:kafka:9092
    reverse:
      - 3000:3000

  frontend:
      forward:
      - 6379:redis:6379
      - 27017:mongo:27017
      - 3306:mysql:3306
      - 9092:kafka:9092
      reverse:
          - 8080:3000
```

Any `okteto up` command will enforce the global forwards are working. 

**Additional context**
In other to avoid running a daemon in the developer machine, `okteto up` could run a goroutine to check if the global forwards are running, and create them otherwise. It requires a goroutine in a loop because if an `okteto up` creates the global forwards, then there is a second `okteto up`, and then, the first `okteto up` exits, the second `okteto up` should regenerate the global forwards.

Also, port collisions should be ignored when another `okteto up` process is running the same okteto manifest, but if the port is in-use by another program, `okteto up` should fail.

**Comment by AdrianPedriza** [comment]
when display forward info, should this global forwards be part of the dev forwards? 
![Captura de pantalla 2022-05-24 a las 15 07 56](https://user-images.githubusercontent.com/22500940/170042326-b20decd7-6245-4609-b9eb-f1e49148f12a.png)
or display it separately ?
![Captura de pantalla 2022-05-24 a las 15 13 51](https://user-images.githubusercontent.com/22500940/170043501-9d61272c-4796-458e-8cfd-43891a204ff2.png)

**Comment by pchico83** [comment]
@AdrianPedriza let's show everything under `Forward`

**Comment by AdrianPedriza** [comment]
Is there any implementation already done to get the information in this case related to another `okteto up` process? some status file or something like that?

**Comment by pchico83** [comment]
@AdrianPedriza every `okteto up` sesion creates a folder with information about the session in `~/.okteto/$OKTETO_NAMESPACE/$OKTETO_MANIFEST_NAME`. Not sure if it will have everything you need

**Comment by AdrianPedriza** [comment]
@pchico83 wouldn't it be better to store this information distinguishing also by app? (`~/.okteto/$OKTETO_NAMESPACE/$OKTETO_APP/$OKTETO_DEV_COMPONENT/..`) maybe you could have dev components belonging to the same namespace with the same name? some typical name like frontend or api

**Comment by pchico83** [comment]
@AdrianPedriza there cannot be two dev containers with the same name in the same namespace. It could be cleaner your way, but I don't think a refactor there makes sense. That path is already used by other integrations like the vscode plugin or the docker extension

**Comment by AdrianPedriza** [comment]
@pchico83 At the moment I can only think of using the format **LOCAL_PORT:SVC_NAME:REMOTE_PORT** because if the name of the service is not specified it is not possible to know exactly from which pod I should make the forward.

**Comment by pchico83** [comment]
> @pchico83 At the moment I can only think of using the format **LOCAL_PORT:SVC_NAME:REMOTE_PORT** because if the name of the service is not specified it is not possible to know exactly from which pod I should make the forward.

@AdrianPedriza right, that would be the only valid format for global services

**Comment by AdrianPedriza** [comment]
@pchico83 I have noticed that if I want to forward a service that is in development mode I am not able to access the exposed service, without any log to inform me about it. Therefore this behavior has also been transferred to global forwarding. Is this behavior expected or should we open an issue?

**Comment by pchico83** [comment]
@AdrianPedriza it isn't expected, but I agree we should handle it in a different issue

**Comment by jmacelroy** [comment]
@AdrianPedriza Can we get a video demo of this feature as it works today? I'd like to understand the use cases it solves better and share the UX with the team for input. 

Also, is there a particular customer @pchico83 that wanted this and is willing to test a beta release? I'm thinking with some of our current port issues of having this not in the main branch until we are more stable but would also like feedback.

**Comment by pchico83** [comment]
> > @pchico83 I have noticed that if I want to forward a service that is in development mode I am not able to access the exposed service, without any log to inform me about it. Therefore this behavior has also been transferred to global forwarding. Is this behavior expected or should we open an issue?
> 
> I wonder about the order of this. I'd prefer to fix the issue before propagating it to a new feature.

@jmacelroy I don't think it's a blocker because the use case for this is not to expose dev container ports, but to expose third party services like databases that every dev container needs to expose locally.

**Comment by pchico83** [comment]
> @AdrianPedriza Can we get a video demo of this feature as it works today? I'd like to understand the use cases it solves better and share the UX with the team for input.
> 
> Also, is there a particular customer @pchico83 that wanted this and is willing to test a beta release? I'm thinking with some of our current port issues of having this not in the main branch until we are more stable but would also like feedback.

@jmacelroy yotpo is waiting for this feature, but it's a common ask from previous customers. If you have two dev containers, and both expose a database in localhost, you need:
- expose them on different ports, which requires different database connection strings configuration
- expose them on the same port. you cannot run both `okteto up` on each dev container at the same time

**Comment by AdrianPedriza** [comment]
@jmacelroy @pchico83 apart from making a demo video for this issue, what would be the best way to document this new feat? just update okteto manifest reference in docs?
